mix.exs

```diff
@@ -56,9 +57,15 @@ defmodule Phoenix.Mixfile do
   end
 
   defp deps do
+    cowboy_version =
```

Comment:
What do you think about pulling this into a function similar to the `lockfile/0`?

```ex
defp cowboy_dep do
  case System.get_env("COWBOY_VERSION") do
    "1" -> {:cowboy, "~> 1.0", optional: true}
    _ -> {:cowboy, github: "ninenines/cowboy", optional: true, override: true}
  end
end
```

Or

```ex
defp cowboy_dep("1"), do: ...
defp cowboy_dep(_), do: ...

defp deps do
  [
    cowboy_dep(System.get_env("COWBOY_VERSION")),
    ...
  ]
end
```